cmd/account.go

```diff
@@ -0,0 +1,94 @@
+// Copyright © 2016 Dropbox, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/dropbox/dropbox-sdk-go-unofficial/users"
+	"github.com/spf13/cobra"
+)
+
+// printFullAccount prints the account details returned by GetCurrentAccount
+func printFullAccount(w *tabwriter.Writer, fa *users.FullAccount) {
+	fmt.Fprintf(w, "Logged in as %s <%s>\n\n", fa.Name.DisplayName, fa.Email)
+	fmt.Fprintf(w, "Account Id:\t%s\n", fa.AccountId)
+	fmt.Fprintf(w, "Account Type:\t%s\n", fa.AccountType.Tag)
+	fmt.Fprintf(w, "Locale:\t%s\n", fa.Locale)
+	fmt.Fprintf(w, "Referral Link:\t%s\n", fa.ReferralLink)
+	fmt.Fprintf(w, "Profile Photo Url:\t%s\n", fa.ProfilePhotoUrl)
+	fmt.Fprintf(w, "Paired Account:\t%t\n", fa.IsPaired)
+	if fa.Team != nil {
+		fmt.Fprintf(w, "Team:\n  Name:\t%s\n  Id:\t%s\n  Member Id:\t%s\n", fa.Team.Name, fa.Team.Id, fa.TeamMemberId)
+	}
+}
+
+// printBasicAccount prints the account details returned by GetAccount
+func printBasicAccount(w *tabwriter.Writer, ba *users.BasicAccount) {
+	fmt.Fprintf(w, "Name:\t%s\n", ba.Name.DisplayName)
+	if !ba.EmailVerified {
+		ba.Email += " (unverified)"
+	}
+	fmt.Fprintf(w, "Email:\t%s\n", ba.Email)
+	fmt.Fprintf(w, "Is Teammate:\t%t\n", ba.IsTeammate)
+	if ba.TeamMemberId != "" {
+		fmt.Fprintf(w, "Team Member Id:\t%s\n", ba.TeamMemberId)
+	}
+	fmt.Fprintf(w, "Profile Photo URL:\t%s\n", ba.ProfilePhotoUrl)
+}
+
+func account(cmd *cobra.Command, args []string) error {
+	w := new(tabwriter.Writer)
+	w.Init(os.Stdout, 4, 8, 1, ' ', 0)
+
+	switch len(args) {
+	case 0:
+		// If no arguments are provided get the current user's account
+		res, err := dbx.GetCurrentAccount()
+		if err != nil {
+			return err
+		}
+		printFullAccount(w, res)
+	case 1:
+		// Otherwise look up an account with the provided ID
+		if len(args[0]) != 40 {
```

Comment:
Hmm, let me see if there's a better way to validate this. String length check seems hackish. We may need to validate the prefix `dbxid` as well.